[Bug] Accepted verify revision leaves stale structured output

## Describe the bug

When a per-task `verify` hook accepts a revised answer, the task's text `output` is replaced but `AgentRunResult.structured` still contains the original pre-revision value.

This makes the two representations disagree and causes downstream tasks using `dependencyPayload: 'structured'` to receive stale data.

## To Reproduce

1. Configure a worker with an `outputSchema`.
2. Run a task through `runTasks()` with `verify: { onDissent: 'revise', maxRounds: 2, ... }`.
3. Have the worker initially return `{"decision":"accept"}`.
4. Have the judge dissent, then have the worker revise to `{"decision":"quarantine"}` and the judge accept.
5. Inspect the completed task result or a dependent task using `dependencyPayload: 'structured'`.

## Current behavior

The completed task contains:

- `output === '{"decision":"quarantine"}'`
- `structured === { decision: 'accept' }`

A dependent task receives the stale `accept` value.

## Expected behavior

An accepted revision should replace the text and parsed representations atomically:

- `output === '{"decision":"quarantine"}'`
- `structured === { decision: 'quarantine' }`

A rejected revision should continue to preserve the original pair.

## Acceptance criteria

- An accepted verify revision updates both `output` and `structured`.
- A downstream task using `dependencyPayload: 'structured'` receives the revised structured value.
- Existing rejected-revision behavior remains unchanged.
- `npm test -w @open-multi-agent/core -- consensus.test.ts` passes.
- `npm run lint -w @open-multi-agent/core` passes.

## Target paths

- `packages/core/src/orchestrator/consensus.ts`
- `packages/core/tests/consensus.test.ts`

## Out of scope

- Public API changes.
- Changes to standalone `runConsensus()` behavior.
- Provider, dependency, documentation, or example changes.

## Open decisions

None.

## Blockers

None.

## Error output

```
AssertionError: expected { decision: 'accept' } to deeply equal { decision: 'quarantine' }
```

## Environment

- OS: macOS
- Node.js version: 24.10.0
- Package version: 1.16.0
- LLM provider: deterministic mock adapters

## Additional context

The issue is observable without network access or provider credentials and is covered by a focused regression test.

--- a/packages/core/src/orchestrator/consensus.ts
+++ b/packages/core/src/orchestrator/consensus.ts
@@ -154,6 +154,8 @@ export interface ConsensusCoreParams {
   readonly prompt: string
   /** Proposed answer to scrutinise (proposer output, or the task result). */
   readonly initialAnswer: string
+  /** Parsed output paired with `initialAnswer`, when the proposer uses an output schema. */
+  readonly initialStructured?: unknown
   /** Usage attributable so far that should be reported back (proposer usage, or zero for the verify hook). */
   readonly initialUsage: TokenUsage
   /** Tokens already spent that count toward the budget but are not re-reported (e.g. prior task usage). */
@@ -181,12 +183,17 @@ export interface ConsensusCoreParams {
   readonly consensusSpan?: TraceSpan
 }
 
+/** Internal consensus result that preserves parsed output across revision rounds. */
+interface ConsensusCoreResult extends ConsensusResult {
+  readonly structured?: unknown
+}
+
 /**
  * Run the judge/refutation loop over a proposed answer: judges run sequentially
  * (so quorum and budget can stop the rest), dissent is recorded to shared memory
  * and trace, and `onDissent` decides whether to revise, reject, or keep.
  */
-export async function runConsensusCore(params: ConsensusCoreParams): Promise<ConsensusResult> {
+export async function runConsensusCore(params: ConsensusCoreParams): Promise<ConsensusCoreResult> {
   const {
     team, prompt, judges, mode, quorum, maxRounds, verdictSchema, onDissent,
     judgePrompt, budget, budgetBaseTokens, reviseProposer, defaults, onTrace, runId,
@@ -196,6 +203,7 @@ export async function runConsensusCore(params: ConsensusCoreParams): Promise<Con
   const sharedMem = team.getSharedMemoryInstance()
 
   let answer = params.initialAnswer
+  let structured = params.initialStructured
   let usage = params.initialUsage
   const dissent: string[] = []
   let rounds = 0
@@ -293,7 +301,10 @@ export async function runConsensusCore(params: ConsensusCoreParams): Promise<Con
       )
       usage = addUsage(usage, r.tokenUsage)
       if (!r.success && executionFailure === undefined) executionFailure = r
-      if (r.success && r.output) answer = r.output
+      if (r.success && r.output) {
+        answer = r.output
+        structured = r.structured
+      }
       if (overBudget() || params.shouldStop?.()) { budgetHit = true; break }
       continue
     }
@@ -304,6 +315,7 @@ export async function runConsensusCore(params: ConsensusCoreParams): Promise<Con
     accepted || (!budgetHit && onDissent === 'keep') ? 'accepted' : 'rejected'
   return {
     answer,
+    ...(structured !== undefined ? { structured } : {}),
     verdict,
     dissent,
     rounds,
@@ -352,6 +364,7 @@ export async function runTaskVerify(
     team,
     prompt: task.description,
     initialAnswer: result.output,
+    initialStructured: result.structured,
     initialUsage: ZERO_USAGE,
     budgetBaseTokens: ctx.cumulativeUsage.input_tokens + ctx.cumulativeUsage.output_tokens,
     judges: verify.judges,
@@ -429,6 +442,7 @@ export async function runTaskVerify(
     result: {
       ...result,
       output: useRevision ? consensus.answer : result.output,
+      structured: useRevision ? consensus.structured : result.structured,
       tokenUsage: addUsage(result.tokenUsage, consensus.tokenUsage),
     },
     verification: {

--- a/packages/core/tests/consensus.test.ts
+++ b/packages/core/tests/consensus.test.ts
@@ -489,6 +489,53 @@ describe('per-task verify hook', () => {
     expect((await mem.read(`worker/task:${taskId}:verdict`))?.value).toBe('accepted')
   })
 
+  it('replaces stale structured output when a verified task accepts a revision', async () => {
+    const worker = captureAdapter((p) =>
+      p.includes('previous answer')
+        ? '{"decision":"quarantine"}'
+        : '{"decision":"accept"}',
+    )
+    const judge = captureAdapter((p) =>
+      p.includes('quarantine') ? ACCEPT : DISSENT,
+    )
+    const consumer = captureAdapter('done')
+    const orch = new OpenMultiAgent()
+    const team = orch.createTeam('t', {
+      name: 't',
+      agents: [{
+        ...agent('worker', worker.adapter),
+        outputSchema: z.object({ decision: z.enum(['accept', 'quarantine']) }),
+      }, agent('consumer', consumer.adapter)],
+    })
+
+    const run = await orch.runTasks(team, [
+      {
+        title: 'verified',
+        description: 'classify the interval',
+        assignee: 'worker',
+        verify: {
+          judges: [agent('judge', judge.adapter)],
+          quorum: 1,
+          maxRounds: 2,
+          onDissent: 'revise',
+        },
+      },
+      {
+        title: 'consume-verdict',
+        description: 'use the verified structured decision',
+        assignee: 'consumer',
+        dependsOn: ['verified'],
+        dependencyPayload: 'structured',
+      },
+    ])
+    const taskId = run.tasks![0]!.id
+
+    expect(run.taskResults?.get(taskId)?.output).toBe('{"decision":"quarantine"}')
+    expect(run.taskResults?.get(taskId)?.structured).toEqual({ decision: 'quarantine' })
+    expect(consumer.prompts[0]).toContain('{"decision":"quarantine"}')
+    expect(consumer.prompts[0]).not.toContain('{"decision":"accept"}')
+  })
+
   it('feeds the prior answer into the revision prompt', async () => {
     const worker = captureAdapter((p) => (p.includes('previous answer') ? 'revised' : 'original'))
     const judge = captureAdapter(DISSENT)

